src/content/blog/astro-550.mdx

```diff
@@ -0,0 +1,137 @@
+---
+title: 'Astro 5.5'
+description: 'Astro 5.5 dives deep with better support for diagramming tools, improved Markdown compatibility, and type-safe sessions!'
+homepageLink:
+  title: 'Astro 5.5'
+  subtitle: 'Available now!'
+publishDate: '2025-03-11'
+authors:
+  - matt
+coverImage: '/src/content/blog/_images/astro-550/blog-post-5-5.webp'
+socialImage: '/src/content/blog/_images/astro-550/og-astro-5-5.webp'
+lang: 'en'
+---
+
+import BlogContentImage from '/src/components/BlogContentImage.astro';
+
+**Astro 5.5 dives deep with better support for diagramming tools, improved Markdown compatibility, and type-safe sessions!**
+
+🧜‍♀️ Dive into a sea of new features with these improvements to Astro:
+
+- [**Better support for diagramming tools in Markdown**](#better-support-for-diagramming-tools-in-markdown)
+- [**Type-safe experimental sessions**](#type-safe-experimental-sessions)
+- [**Improved heading ID compatibility**](#improved-heading-id-compatibility)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Better support for diagramming tools in Markdown
+
+Tools like Mermaid and D2 let you define charts and diagrams in code blocks in Markdown, which are then rendered inside the content. Until now, using these in Astro could be tricky, because syntax highlighting would interfere with how the code blocks are rendered. Astro 5.5 introduces a new `excludeLangs` option in your Markdown configuration that makes it possible to skip syntax highlighting for specific language code blocks. This unlocks the ability to use tools like these without disabling syntax highlighting for other blocks.
```

Comment:
```suggestion
Tools like Mermaid and D2 let you define charts and diagrams in code blocks in Markdown, which are then rendered inside the content. Until now, using these in Astro could be tricky, because Astro's default syntax highlighting would interfere with how the code blocks were rendered.

Astro 5.5 introduces a new `excludeLangs` option in your Markdown configuration that makes it possible to skip syntax highlighting for specific language code blocks. This unlocks the ability to use tools like these without disabling syntax highlighting for other blocks.
```